### PR TITLE
add test for ComponentFinder.watch

### PR DESCRIPTION
**Pull request by chetverikov**

### CODE REVIEW

File: lib/finders/ComponentFinder.js [hunk 1]
```diff
@@ -127,15 +127,26 @@ class ComponentFinder extends events.EventEmitter {
...
-			Object.keys(this._foundComponentsByDirs), CHOKIDAR_OPTIONS
-		)
+		const readyDefers = [Promise.defer(), Promise.defer()];
+
+		this._readyWatchers = Promise
+			.all([readyDefers[0].promise, readyDefers[1].promise])
+			.then(data => {
+				this.emit('watchReady');
+				return data;
+			});
```
Comment (pragmadash) [inline lib/finders/ComponentFinder.js]: Please avoid using `defer`. It's not a part of Promise spec and we can easily use `new Promise((resolve, reject) => ...)` here.

File: lib/finders/ComponentFinder.js [hunk 2]
```diff
@@ -231,6 +239,25 @@ class ComponentFinder extends events.EventEmitter {
...
 				this.emit('unlink', component);
 			});
+
+		return this._readyWatchers;
+	}
+
+	/**
+	 * Close watchers
+	 *
+	 * @return {Promise} Promise resolved when all watcher will be close
```
Comment (pragmadash) [inline lib/finders/ComponentFinder.js]: Promise of closed watchers.

File: lib/finders/ComponentFinder.js [hunk 3]
```diff
@@ -231,6 +239,25 @@ class ComponentFinder extends events.EventEmitter {
...
+	 */
+	closeWatch() {
+		if (this._readyWatchers) {
+			const promise = this._readyWatchers;
+
+			this._readyWatchers = null;
+
+			return promise.then(watchers => watchers.map(watcher => watcher.close()));
```
Comment (pragmadash) [inline lib/finders/ComponentFinder.js]: Why are you using `map` here? I bet the `close` method does not return a Promise here, right?

File: lib/promises/promiseHelper.js [hunk 4]
```diff
@@ -5,24 +5,37 @@ module.exports = {
 	/**
 	 * Converts a method with a callback to the method that returns a promise.
 	 * @param {Function} methodWithCallback The method with a callback.
+	 * @param {Object} options
+	 * @param {*} options.thisArg Context for execute methodWithCallback
+	 * @param {Boolean} options.ignoreError Not reject on fist argument
 	 * @returns {Function} The method that returns a promise.
 	 */
-	callbackToPromise: methodWithCallback => {
+	callbackToPromise: (methodWithCallback, options) => {
```
Comment (pragmadash) [inline lib/promises/promiseHelper.js]: We don't need to change this method. Let's keep it as it is.

File: package.json [hunk 5]
```diff
@@ -1,112 +1,112 @@
...
-		"eslint": "^2.2.0"
-	},
-	"engines": {
-		"node": ">=4"
-	},
-	"scripts": {
-		"test": "make"
-	},
-	"license": "MIT"
+  "name": "catberry",
```
Comment (pragmadash) [inline package.json]: Let's keep the indentation as it was.

File: package.json [hunk 6]
```diff
@@ -1,112 +1,112 @@
...
+    "mkdirp": "~0.5.1",
+    "morphdom": "^1.1.2",
+    "pretty-hrtime": "^1.0.2",
+    "promise": "^7.1.1",
+    "uglify-js": "^2.6.2",
+    "uuid": "^2.0.1",
+    "watchify": "^3.7.0"
+  },
+  "devDependencies": {
+		"fs-extra": "^0.30.0",
```
Comment (pragmadash) [inline package.json]: New dependency? Why?

**Comment by pragmadash** [comment]
Too many changes in the source code for adding tests. I would prefer just to add a new event `startWatchingComponents` in the EventBus and keep the rest of it as it is.

**Comment by pragmadash** [comment]
@chetverikov any updated on that? Should I continue working on this branch or you have something to push?

**Comment by chetverikov** [comment]
@pragmadash Yes, you can continue working on this branch.

**Pull request closed without merging**